python/lib/cloudutils/utilities.py

```diff
@@ -134,17 +134,23 @@ def __init__(self):
                 self.distro = "Ubuntu"
             else:
                 raise UnknownSystemException(distributor)
-        else: 
+        elif os.path.exists("/etc/os-release"):
+            version = open("/etc/os-release").readline()
+            if version.find("openSUSE") != -1:
+                self.distro = "openSUSE"
+            else:
+                raise UnknownSystemException(distributor)
```

Comment:
```suggestion
```
since `distributor` is not set and those `raise` from two lines below will be enough

Reply:
@davidjumani you still need to remove this, since `distributor` is not defined and those would raise an additional `NameError` exception, but not the expected `UnknownSystemException`
```
>>> raise UnknownSystemException(distributor)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'distributor' is not defined
```

Reply:
@davidjumani you still need to remove this, since `distributor` is not defined and those would raise an additional `NameError` exception, but not the expected `UnknownSystemException`
```
>>> raise UnknownSystemException(distributor)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'distributor' is not defined
```

Reply:
```suggestion
            else:
                distributor=`grep -e "^NAME=" /etc/os-release | awk -F\" '{print $2}'` or None
                raise UnknownSystemException(distributor)
```
or get `distributor` from `/etc/os-release`